_data-prepper/pipelines/configuration/sources/dynamo-db.md

```diff
@@ -88,8 +88,107 @@ Option | Required | Type | Description
+        ],
+        "Resource": [
+          "arn:aws:dynamodb:us-east-1:{account-id}:table/my-table"
+        ]
+      },
+       {
+            "Sid": "allowRunExportJob",
+            "Effect": "Allow",
+            "Action": [
+                "dynamodb:DescribeContinuousBackups",
+                "dynamodb:ExportTableToPointInTime"
+            ],
+            "Resource": [
+                "arn:aws:dynamodb:us-east-1:{account-id}:table/my-table"
+            ]
+        },
+        {
+            "Sid": "allowCheckExportjob",
+            "Effect": "Allow",
+            "Action": [
+                "dynamodb:DescribeExport"
+            ],
+            "Resource": [
+                "arn:aws:dynamodb:us-east-1:{account-id}:table/my-table/export/*"
+            ]
+        },
+        {
+            "Sid": "allowReadFromStream",
+            "Effect": "Allow",
+            "Action": [
+                "dynamodb:DescribeStream",
+                "dynamodb:GetRecords",
+                "dynamodb:GetShardIterator"
+            ],
+            "Resource": [
+                "arn:aws:dynamodb:us-east-1:{account-id}:table/my-table/stream/*"
+            ]
+        },
+        {
+            "Sid": "allowReadAndWriteToS3ForExport",
+            "Effect": "Allow",
+            "Action": [
+                "s3:GetObject",
+                "s3:AbortMultipartUpload",
+                "s3:PutObject",
+                "s3:PutObjectAcl"
+            ],
+            "Resource": [
+                "arn:aws:s3:::my-bucket/*"
+            ]
+        }
+    ]
+}
+```
+
+When performing an export, `"Sid": "allowReadFromStream"` section is not required. If only reading from DynamoDB streams, the 
+`"Sid": "allowReadAndWriteToS3ForExport"`, `"Sid": "allowCheckExportjob"`, ` "Sid": "allowRunExportJob"` sections are not required.
+
+## Metrics
+
+The `dynamodb` source includes the following metrics:
 
+### Counters
 
+* `exportJobSuccess` - The number of export jobs that have been submitted successfully.
+* `exportJobFailure` - The number of export job submission attempts that have failed.
+* `exportS3ObjectsTotal` - The total number of export data files found in S3 for an export.
+* `exportS3ObjectsProcessed` - The total number of export data files that have been processed successfully from S3.
+* `exportRecordsTotal` - The total number of records found in the entire export.
+* `exportRecordsProcessed` - The total number of export records that have been processed successfully.
+* `exportRecordsProcessingErrors` - The number of export record processing errors.
+* `changeEventsProcessed` - The number of change events processed from DynamoDB streams.
+* `changeEventsProcessingErrors` - The number of processing errors for change events from DynamoDB streams.
+* `shardProgress` - The incremented shard progress when DynamoDB streams are being read correctly. If this is `0` for a certain period of time, then there is a problem with the pipeline that has streams enabled.
```

Comment:
Should we specify what we mean by "a certain period of time", or do we mean something like "for any length of time"?

Reply:
"For any significant length of time?"

Reply:
I think its really dependent on the user's configuration and what their own expected shard progress time is.

Reply:
Ok, then "For any significant length of time" works.